test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs

```diff
@@ -93,68 +93,344 @@ await foreach (Page<Product> page in allPages.AsPages())
         });
 
         [Test]
-        public Task RawLRO() => Test(async (host) =>
+        [TestCase(true)]
+        [TestCase(false)]
+        public Task RawLRO(bool nestedScopeSuppression) => Test(async (host) =>
         {
             using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
             CollectionAssert.IsEmpty(diagnosticListener.Scopes);
 
-            Operation<BinaryData> result = await new DPGClient(Key, host, null).LroAsync(WaitUntil.Started, "raw");
+            Operation<BinaryData> result = await new DPGClient(Key, host, null, nestedScopeSuppression).LroAsync(WaitUntil.Started, "raw");
             diagnosticListener.AssertAndRemoveScope("DPGClient.Lro");
             CollectionAssert.IsEmpty(diagnosticListener.Scopes);
 
             await result.WaitForCompletionAsync();
             diagnosticListener.AssertAndRemoveScope("DPGClient.Lro.WaitForCompletion");
-            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            if (nestedScopeSuppression)
+            {
+                CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            }
 
             JsonData responseBody = JsonData.FromBytes(result.Value.ToMemory());
             Assert.AreEqual("raw", (string)responseBody["received"]);
         });
 
         [Test]
-        public Task HandwrittenModelLro() => Test(async (host) =>
+        [TestCase(true)]
+        [TestCase(false)]
+        public Task RawLROResponseOperationOfT(bool nestedScopeSuppression) => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            Operation<BinaryData> result = await new DPGClient(Key, host, null, nestedScopeSuppression).LroAsync(WaitUntil.Started, "raw");
+            diagnosticListener.AssertAndRemoveScope("DPGClient.Lro");
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            await result.WaitForCompletionResponseAsync();
+            diagnosticListener.AssertAndRemoveScope("DPGClient.Lro.WaitForCompletion");
+            if (nestedScopeSuppression)
+            {
+                CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            }
+
+            JsonData responseBody = JsonData.FromBytes(result.Value.ToMemory());
+            Assert.AreEqual("raw", (string)responseBody["received"]);
+        }, ignoreScenario: true);
```

Comment:
Curious: why `ignoreScenario`?

Reply:
I added it because the tests won't pass unless they either have ignoreScenario set to true or they're set [in the test server](https://github.com/Azure/autorest.testserver/blob/main/src/test-routes/dpg/customization.ts#L4), it seems to me like adding these scenarios to the test server would imply that the other languages should also test these cases as well, but I believe for now the plan was to only add these tests to .NET?